nixos/modules/services/networking/smartdns.nix

```diff
@@ -0,0 +1,66 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.smartdns;
+
+  confFile = pkgs.writeText "smartdns.conf" ''
+    bind :${toString cfg.bindPort}
+    cache-size ${toString cfg.cacheSize}
+    ${cfg.servers}
+    ${cfg.extraConfig}
+  '';
+in {
+  options = {
+
+    services.smartdns = {
+
+      enable = mkEnableOption "SmartDNS DNS server";
+
+      servers = mkOption {
+        type = types.lines;
+        default = "";
+        description =
+          "List of different DNS servers. Support UDP, TCP, TLS and HTTPS to query. See SmartDNS's readme for more info.";
+      };
+
+      cacheSize = mkOption {
+        type = types.int;
+        default = 512;
+        description = "Domain name result cache number. Default 512.";
+      };
+
+      bindPort = mkOption {
+        type = types.int;
+        default = 53;
+        description = "DNS listening port number. Default port 53.";
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = "";
+        description =
+          "Any extra config that will be appended in configuration file";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.smartdns = {
```

Comment:
Could you use upstream service file instead using `systemd.packages` option?

Reply:
> Could you use upstream service file instead using `systemd.packages` option?

I will see it later. Is there any example that resolves like what you said? I could use it as a reference.

Reply:
Just search for `systemd.packages` in pull requests. Example: https://github.com/NixOS/nixpkgs/pull/59480

Reply:
Thanks